Add docs on database read options

markcmiller86 opened the pull request:
I started down this path primarily to avoid putting the ffp plugin info in email. Once I did, I decided to tackle a few other database read option cases as well.

You can see *rendered* docs [here](https://visit-sphinx-github-user-manual.readthedocs.io/en/doc-mcm86-25feb20-db-read-options/gui_manual/WorkingWithFiles/Read_Write_Options.html)

markcmiller86 committed:
adding section on read options

markcmiller86 committed:
adding info on database read options

rusu24edward commented on src/doc/gui_manual/WorkingWithFiles/Read_Write_Options.rst:
```diff
@@ -0,0 +1,184 @@
+
+Database Read and Write Options
+-------------------------------
+Several database plugins have options that effect reading and/or writing with that
```
I believe the verb should be 'affect' not 'effect'. I have to look it up every time to double check. Here is a resource: https://www.dictionary.com/e/affect-vs-effect/

rusu24edward commented on src/doc/gui_manual/WorkingWithFiles/Read_Write_Options.rst:
```diff
@@ -0,0 +1,184 @@
+-------------------------------
+Several database plugins have options that effect reading and/or writing with that
+format. These are described in this section. Alternatively, in some cases, the
+behavior of a database plugin may be effected by enviornment variables.
```
Same thing here with affect/effect

rusu24edward commented on src/doc/gui_manual/WorkingWithFiles/Read_Write_Options.rst:
```diff
@@ -0,0 +1,184 @@
+format. These are described in this section. Alternatively, in some cases, the
+behavior of a database plugin may be effected by enviornment variables.
+
+Chombo
```
The resulting formatting throws me off a little. So when you have this, do you mean that the property "Detect Compound Variables" is the same for both Chombo and Exodus? Or is this a stub for the chombo database to be filled out later?

markcmiller86 replied:
Well, its because I left some sections blank but didn't make that clear. I've added some boilerplate text in those blank sections to help clarify.

rusu24edward commented on src/doc/gui_manual/WorkingWithFiles/Read_Write_Options.rst:
```diff
@@ -0,0 +1,184 @@
+~~~~~~
+   
+
+Detect Compound Variables
```
This is a cool feature!

markcmiller86 replied:
Thanks. Now that I think about it...this really should be something we offer *above* all the plugins, instead of coding it in individual plugins.

rusu24edward commented on src/doc/gui_manual/WorkingWithFiles/Read_Write_Options.rst:
```diff
@@ -0,0 +1,184 @@
+specific values.
+
+The ALEGRA nameschemes for volume fraction and material specific variables are
+``"@%s@n?'&VOLFRC_%d&n&':'VOID_FRC':@"`` and ``"@%V_%d@n"``.
```
Not sure if this is intended to be formatted by sphinx in some way, but it pretty much turned out the way it is written here. Just wanted to confirm that was intended

markcmiller86 replied:
Yes, those are *literal* namescheme strings.

rusu24edward commented on src/doc/gui_manual/WorkingWithFiles/Read_Write_Options.rst:
```diff
@@ -0,0 +1,184 @@
+to improve its behavior and performance. It will do so by loading the STRIPACK library as
+a dynamicaly loaded shared library *when* VisIt_ can find it. VisIt_ will find the
+STRIPACK library if it is available in the VisIt_ installation's top-level ``lib``
+directory (typically something like) or if the enviornment variable 
```
Not sure if I'm misreading this somehow, but I'm getting a bit confused with the 'typically something like' part here.

rusu24edward commented on src/doc/gui_manual/WorkingWithFiles/Read_Write_Options.rst:
```diff
@@ -0,0 +1,184 @@
+In client/server mode, the STRIPACK library must be installed on both the client
+and the server.
+
+The `build_visit <Building>` tool can be used to download, build and install the
```
Not sure if `<Building>` is also format that that sphinx should convert, but it appears to fail here

rusu24edward commented on src/doc/gui_manual/WorkingWithFiles/Read_Write_Options.rst:
```diff
@@ -0,0 +1,184 @@
+
+.. code-block:: shell
+
+    env FCFLAGS="-fdefault-real-8 -fdefault-double-8 -shared -fPIC" STRIPACK_INSTALL_DIR=/usr/local/visit/lib ./build_visit --fortran --no-visit --no-thirdparty --thirdparty-path /dev/null --no-zlib --stripack
```
I tend to favor line breaks in code examples so that the readers don't have to scroll horizontally.

rusu24edward commented on src/doc/gui_manual/WorkingWithFiles/Read_Write_Options.rst:
```diff
@@ -0,0 +1,184 @@
+Because STRIPACK is non-BSD licensed software, part of the ``build_visit``
+process for installing it is to accept the STRIPACK license terms.
+
+H5Part
```
Same question as my note on line 9

rusu24edward commented on src/doc/gui_manual/WorkingWithFiles/Read_Write_Options.rst:
```diff
@@ -0,0 +1,184 @@
+because that will avert the plugin having to read all lines of the input before
+understanding the material configuration. However, if the user does not know
+the number of materials, enter ``-1`` here and the plugin will search for
+all information related the material configuration during the *open*. This will
```
Need 'related to'

rusu24edward commented on src/doc/gui_manual/WorkingWithFiles/Read_Write_Options.rst:
```diff
@@ -0,0 +1,184 @@
+``Never`` mean to always *ignore* or never *ignore* the extents data and
+``Auto`` means to ignore extents data for files written by data producers known
+to have issues with extents data in the past. The ``Undef`` setting is to deal
+with cases where users may have `saved settings <How to Save Settings>` with
```
Looks like sphinx didn't catch this link either

markcmiller86 committed:
fixing @rusu24edward requests

markcmiller86 committed:
fix missing example paranthetical

rusu24edward reviewed (commented):
Two final thoughts, just comments though, not necessarily needed to change.
1. Instead of stubbing the missing sections, it might be better to comment them from the rst file and add tickets for each. That way, the documentation still looks clean, and we as developers have a better sense of what needs to be filled.
2. If there are "features/options" shared among multiple databases, how can we present this without duplicated the docs? If there are not many shared features, then this probably isn't a problem. But if so, perhaps a table/matrix at the top of the page showing which feature and which plugin supports it.

rusu24edward commented:
Also, can you regen the documentation? I just want to double check the links

markcmiller86 commented:
> Also, can you regen the documentation? I just want to double check the links

It should be re-gening automatically on RTD and available via the same link as at top. But, might be a 10-20 min delay depending on other factors.

brugger1 commented on src/doc/gui_manual/WorkingWithFiles/Read_Write_Options.rst:
```diff
@@ -0,0 +1,212 @@
+Database Write Options
+----------------------
+
+This section incomplete.
```
At the top you have Database Read and Write Options at the same level as the Database Write Options here. Should the top be Database Read Options or should this section go away?

markcmiller86 committed:
adding zipwrapper stuff

markcmiller86 commented:
> Two final thoughts, just comments though, not necessarily needed to change.
> 
> 1. Instead of stubbing the missing sections, it might be better to comment them from the rst file

I removed them.

> and add tickets for each.

I added a *single* ticket

> That way, the documentation still looks clean, and we as developers have a better sense of what needs to be filled.
> 2. If there are "features/options" shared among multiple databases, how can we present this without duplicated the docs? If there are not many shared features, then this probably isn't a problem. But if so, perhaps a table/matrix at the top of the page showing which feature and which plugin supports it.

I don't think this case crops up yet. But, if we ever did add the code to support options *above* any given database plugin, it would be natural to then document this in a place in the documentation that is *above* any particular plugin and I think that would work fine even with the way we manage the docs now.

markcmiller86 committed:
Merge branch 'develop' into doc/mcm86/25Feb20/db-read-options

rusu24edward commented on src/doc/gui_manual/WorkingWithFiles/Read_Write_Options.rst:
```diff
@@ -192,21 +147,46 @@ order to handle them VisIt_ needs to be forced to go searching for their
+Don't atexit()
+""""""""""""""
+Ordinarily, when VisIt_ exits, it will remove any decompressed files it
+left around from invokations of ZipWrapper's decompression logic. This
```
invokation -> invocation

rusu24edward commented on src/doc/gui_manual/WorkingWithFiles/Read_Write_Options.rst:
```diff
@@ -192,21 +147,46 @@ order to handle them VisIt_ needs to be forced to go searching for their
-This section incomplete.
+Max. # decompressed files
+"""""""""""""""""""""""""
+Specifies the maximum number of decompressed files that can be in existance
```
existance -> existence

markcmiller86 commented:
Oops. I'll fix those.
